static/app/components/organizations/projectSelector/index.tsx

```diff
@@ -393,21 +393,27 @@ function ProjectSelector({
               onApply={() => handleUpdate(actions)}
               onShowAllProjects={() => {
                 handleQuickSelect({id: ALL_ACCESS_PROJECTS.toString()});
-                actions.close();
                 trackAdvancedAnalyticsEvent('projectselector.multi_button_clicked', {
                   button_type: 'all',
                   path: getRouteStringFromRoutes(router.routes),
                   organization,
                 });
+
+                // Close action triggers onClose which needs to run on next render
+                // with updated state to work correctly
+                setTimeout(actions.close);
```

Comment:
I don't like having to do this since it's clearly a hack 😞 

The other options are:

- Modifying the behavior of our `AutoComplete` component to not trigger `onClose` when calling from `actions.close()`. Not sure I like this since you might expect that to happen in other use cases
- Somehow modifying the `onClose` to accept new arguments that tell it why it was called. Then the `onClose` here could return early if it was called manually. But the typing for the `onClose` is a bit strange: `onClose?: (...args: Array<any>) => void;` so that could have other consequences.

Reply:
Could we expand the comment for *why* it needs to run on the next render?

Reply:
Updated with a more descriptive comment